MekHQ/src/mekhq/campaign/mission/resupplyAndCaches/Resupply.java

```diff
@@ -0,0 +1,2371 @@
+        panel.add(imageLabel, BorderLayout.CENTER);
+        panel.add(textPane, BorderLayout.SOUTH);
+
+        // Create a custom dialog
+        JDialog dialog = new JDialog();
+        dialog.setTitle(title);
+        dialog.setLayout(new BorderLayout());
+
+        // Create an accept button and add its action listener.
+        JButton acceptButton = new JButton(resources.getString("convoyConfirm.text"));
+        acceptButton.addActionListener(e -> dialog.dispose());
+
+        // Create a panel for buttons and add buttons to it
+        JPanel buttonPanel = new JPanel();
+        buttonPanel.add(acceptButton);
+
+        // Add the original panel and button panel to the dialog
+        dialog.add(panel, BorderLayout.CENTER);
+        dialog.add(buttonPanel, BorderLayout.SOUTH);
+
+        dialog.setResizable(false);
+        dialog.pack();
+        dialog.setModal(true);
+        dialog.setLocationRelativeTo(null);
+        dialog.setVisible(true);
+    }
+
+    /**
+     * Calculates the convoy count and total cargo capacity of the player's convoys.
+     */
+    private void calculatePlayerConvoyValues() {
+        playerConvoys = new HashMap<>();
+        totalPlayerCargoCapacity = 0;
+
+        for (StrategicFormation formation : campaign.getStrategicFormationsTable().values()) {
+            Force force = campaign.getForce(formation.getForceId());
+
+            if (force == null) {
+                continue;
+            }
+
+            double cargoCapacitySubTotal = 0;
+            if (force.isConvoyForce()) {
+                boolean hasCargo = false;
+                for (UUID unitId : force.getAllUnits(false)) {
+                    try {
+                        Unit unit = campaign.getUnit(unitId);
+                        Entity entity = unit.getEntity();
+
+                        if (unit.isDamaged()
+                            || !unit.isFullyCrewed()
+                            || isProhibitedUnitType(entity, true)) {
+                            continue;
+                        }
+
+                        double individualCargo = unit.getCargoCapacity();
+
+                        if (!hasCargo && individualCargo > 0) {
+                            hasCargo = true;
+                        }
+
+                        cargoCapacitySubTotal = individualCargo;
+                    } catch (Exception ignored) {
+                        // If we run into an exception, it's because we failed to get Unit or Entity.
+                        // In either case, we just ignore that unit.
+                    }
+                }
+
+                if (hasCargo) {
+                    if (cargoCapacitySubTotal > 0) {
+                        totalPlayerCargoCapacity += cargoCapacitySubTotal;
+                        playerConvoys.put(force, cargoCapacitySubTotal);
+                    }
+                }
+            }
+        }
+    }
+
+    public void createPlayerConvoyOptionalDialog() {
```

Comment:
Really, this need to be split into many small classes, one for each individual part of it. The data has to be split from it too.
This is better done now than later, because later it will be too much of a pain to do properly, and its only going to grow more and more. Lets keep the new stuff neat and not repeat the errors of the legacy code.